mlrun/secrets.py

```diff
@@ -63,3 +70,43 @@ def get_all(self):
     def to_serial(self):
         # todo: use encryption
         return [{'kind': 'inline', 'source': self._secrets.copy()}]
+
+
+def new_key():
+    """Return a new encryption key"""
+    return Fernet.generate_key()
+
+
+def _enc(value, key):
+    enc = Fernet(key)
+    data = enc.encrypt(pickle.dumps(value))
+    return data.decode('utf-8')
+
+
+def encrypt_dict(d: dict, key=None):
+    """Encrypt values in a dictionary"""
+    enc_key = _resolve_key(key)
+    return {key: _enc(val, enc_key) for key, val in d.items()}
+
+
+def _dec(value, key):
+    if isinstance(value, str):
+        value = value.encode('utf-8')
+    dec = Fernet(key)
+    return pickle.loads(dec.decrypt(value))
+
+
+def decrypt_dict(d: dict, key=None):
+    """Decrypt values in a dictionary"""
+    enc_key = _resolve_key(key)
+    return {key: _dec(val, enc_key) for key, val in d.items()}
+
+
+def _resolve_key(key):
+    if key:
+        return key
+
+    key = environ.get(_env_enc_key)
+    if not key:
+        raise ValueError(f'key is None and {_env_enc_key} not set')
+    return key.encode('utf-8')
```

Comment:
@tebeka we want the key to be read from a file (i.e. env var = filepath w a default value), also we can assume that secrets are strings (i.e. no need for pickle)